### PR TITLE
TYP: internals.construction

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
Would it be OK to wait on https://github.com/pandas-dev/pandas/pull/39991 for this?

**Comment by jbrockmendel** [comment]
> Would it be OK to wait on #39991 for this?

should be OK

**Comment by jorisvandenbossche** [comment]
Thanks, https://github.com/pandas-dev/pandas/pull/39991 is merged now

**Comment by jbrockmendel** [comment]
updated, should fix the mypy complaint that made it into master

File: pandas/core/frame.py [hunk 1]
```diff
@@ -1939,12 +1939,11 @@ def from_records(
...
                         arrays.append(v)
 
-                arrays, arr_columns = reorder_arrays(arrays, arr_columns_list, columns)
+                arr_columns = Index(arr_columns_list)
+                arrays, arr_columns = reorder_arrays(arrays, arr_columns, columns)
 
         elif isinstance(data, (np.ndarray, DataFrame)):
             arrays, columns = to_arrays(data, columns)
-            if columns is not None:
-                columns = ensure_index(columns)
```
Comment (jorisvandenbossche) [inline pandas/core/frame.py:1947]: This is ensured by `to_arrays` already?

File: pandas/core/internals/construction.py [hunk 2]
```diff
@@ -91,7 +92,7 @@ def arrays_to_mgr(
     dtype: Optional[DtypeObj] = None,
     verify_integrity: bool = True,
     typ: Optional[str] = None,
-):
+) -> Union[BlockManager, ArrayManager]:
```
Comment (jorisvandenbossche) [inline pandas/core/internals/construction.py:95]: There is a typing alias for this union (`Manager`, I think)

File: pandas/core/internals/construction.py [hunk 3]
```diff
@@ -140,9 +141,8 @@ def rec_array_to_mgr(
     fdata = ma.getdata(data)
     if index is None:
         index = _get_names_from_index(fdata)
-        if index is None:
-            index = ibase.default_index(len(data))
-    index = ensure_index(index)
+    else:
+        index = ensure_index(index)
```
Comment (jorisvandenbossche) [inline pandas/core/internals/construction.py:145]: Should this be left unindented (without `else`)? `_get_names_from_index` doesn't necessarily returns an index?

Comment (jorisvandenbossche) [reply pandas/core/internals/construction.py:145]: Ah, I see you changed that below ;)

File: pandas/core/frame.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/frame.py:1947]: yes

File: pandas/core/internals/construction.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/internals/construction.py:95]: mind keeping this for a follow-up?  id like to get the CI back to green

Comment (jorisvandenbossche) [reply pandas/core/internals/construction.py:95]: CI checks are also still failing on this PR

**Comment by jbrockmendel** [comment]
this now gets the code checks passing (just the bs4 thing left)

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks!

**Pull request merged**